docs/core/concepts/flows.md

```diff
@@ -192,3 +192,42 @@ state_handler(flow: Flow, old_state: State, new_state: State) -> State
 The handler is called anytime the flow's state changes, and receives the flow itself, the old state, and the new state. The state that the handler returns is used as the flow's new state.
 
 If multiple handlers are provided, they are called in sequence. Each one will receive the "true" `old_state` and the `new_state` generated by the previous handler.
+
+## Terminal State Handlers
+
+Flows allow users to provide custom logic for determining the final State of a Flow.
+
+Unlike normal state handlers, the terminal handler has access to the final states of all tasks in the flow.
+
+Terminal state handlers must have the following signature:
+
+```python
+terminal_state_handler(flow: Flow, state: State, task_states: Dict[Task, State]) -> Optional[State]
+```
+
+`flow` is the current Flow
+`state` is the current state of the Flow
+`task_states` contains states for Flow Tasks
+
+For example, you may want to determine which reference tasks that have failed and add them to your final state message.
+
+This is particularly useful for Cloud Hooks that report the state message.
```

Comment:
```suggestion
For example, you may want to determine which reference tasks have failed and add them
to your final state message. This could be used to improve the message sent out by Cloud
Hooks that report the flow run's final state.
```